### ISSUE TITLE
Multilingual Support

### ISSUE THREAD

**Issue by zakki** [slp-rl/slamkit#3]
I trained Qwen2.5-0.5B on a Japanese dataset using an RTX 4090, but the results were poor.
Following extract_features.py and eval.py, even when generating speech from tokens extracted from the training data—rather than tokens generated by an LLM—the output sounds unnatural in Japanese.

Are the training of the feature extractor and the vocoder outside the scope of this project?

**Comment by gallilmaimon** [comment]
Hey,

Yes, that is most likely an issue with the Vocoder as it was only trained on a single speaker (LJ) English only dataset.

We don't provide training code for training the feature extractor and vocoder as part of this codebase, because we want to have it focused on the SLM itself (at least for now), but we are happy to add support for training SLMs over different tokenisers and with different vocoders.

If you wanted you could try train your own vocoder over the existing multi-lingual hubert (and maybe re-fitting the kmeans). The training is out of scope for this repo, but anything trained with `textlesslib` should be easy to load here:
**Encoder** - can be exported like so https://github.com/huggingface/transformers/blob/dcbdf7e962c4b36140cc9ee76f870016121e69e5/src/transformers/models/hubert/convert_hubert_original_pytorch_checkpoint_to_pytorch.py#L249

**KMeans** - can probably be loaded directly with a different kmeans path - https://github.com/slp-rl/slamkit/blob/ca7fb113409196383eaf1b30fef5ca24c3be8462/slamkit/feature_extractor/hubert_feature_extractor.py#L18

**Vocoder** - we can natively load any `textlesslib` or we are happy to help you in the process of adding other trained ones to the repository.



Final comment - if you wanted to test the modelling abilities of the SLM independent of the vocoder you could do that with modelling metrics like sWUGGY, Salmon etc - https://github.com/slp-rl/slamkit/blob/ca7fb113409196383eaf1b30fef5ca24c3be8462/slamkit/metric/modelling_metric.py#L79
These do not depend on the vocoder, but I am unaware of a Japanese version of these benchmarks.

Hope that helps :)

**Comment by zakki** [comment]
Thank you.
I found an open hubert variant trained on Japanese data, but I couldn't find any k-means or vocoder.
If I don't find them after a little more searching, I'll look into textlesslib's training methods.
